team/TESTING.md

```diff
@@ -0,0 +1,33 @@
+# Testing Approach
```

Comment:
I agree comprehensive component, integration, end-to-end tests I feel take a lot of time just due to the fact that there are lot of moving parts. I think we would have to develop some "infrastructure" to set up a base for these higher level tests to make it easier when writing one. Of course this would take a lot of time. For meaningful end-to-end tests/integration test I feel there should be no mocking going on, since we're going to be testing with a "near-production" deployment. This would be done with something like Selenium or Puppeteer for testing the entire app. And we would have create some more infrastructure to create a "test" database and whatnot. Obviously it's quite a lot for little benefit since we're on such a short time frame. Simple manual approach by checking on our local development environment during PRs seems to work well for now. We could also "automate" and have the PR branch be deployed in another separate "QA" app deployment and database.

Making unit tests still takes up good amount of time anyways (but it's relatively more straightforward and simple since we're only testing an isolated part of source code/single component rather than a page).

Reply:
yeah completely agree here. I think a full end-to-end test might be more useful than the intermediate component tests since they kind of encompass the scope of component tests as well (for an end-to-end test to work the page components need to properly compose their subcomponents).